src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines =
+                type == Bulldozer.class ? new BulldozerProducer().get()
+                : type == Excavator.class ? new ExcavatorProducer().get()
+                : type == Truck.class ? new TruckProducer().get()
+                : Collections.emptyList();
+        return new ArrayList<>(machines);
```

Comment:
This code must be a good example of polymorphism in java. Let's create a var of a type MachineProducer and call the method `get` only once.